Ignores allure-results in git.

hostep opened the pull request:
### Description (*)
When running unit tests on a clone of this repo, it generates some files which should be ignored in git.

I'm not sure if we would need to ignore the entire `dev/tests/unit/var/` directory, or only `dev/tests/unit/var/allure-results`? I've opted for the latter here.

### Fixed Issues (if relevant)
None that I could find.

### Manual testing scenarios (*)
1. Clone this repo (2.3-develop branch)
2. Run `composer install`
3. Run `cd dev/tests/unit`
4. Run `../../../vendor/bin/phpunit ../../../setup/src/Magento/Setup/Test/Unit/`
5. Run `cd ../../../`
5. Run `git status`
6. It is expected that nothing shows up, but the directory `dev/tests/unit/var/allure-results` shows up containing a whole bunch of generated xml files.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

hostep committed:
Ignores allure-results in git.

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5055](https://jira.corp.magento.com/browse/ENGCOM-5055) has been created to process this Pull Request

hostep commented:
@ihor-sviziev: I've updated my PR, there were more files to be ignored.
Now I ran: `bin/magento bin/magento dev:tests:run` which runs more then only the unit tests.
This generated a bunch more `allure-results` directories, so I've gone ahead and ignored all those others as well.

In the `.gitignore` file of the static tests, I also updated the other lines to start with a `/`, so those lines are consistent with most other `.gitignore` files in the `dev` directory.

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5055](https://jira.corp.magento.com/browse/ENGCOM-5055) has been created to process this Pull Request

stoleksiy commented:
:heavy_check_mark: QA Passed 

**Before:**

![before](https://user-images.githubusercontent.com/44244273/57221051-4f0ede80-7006-11e9-89f8-8a566b40331a.png)

**After:**

![after](https://user-images.githubusercontent.com/44244273/57221058-559d5600-7006-11e9-87b9-5aebdf56b828.png)
